src/Symfony/Component/Form/Console/Descriptor/JsonDescriptor.php

```diff
@@ -20,6 +20,16 @@
  */
 class JsonDescriptor extends Descriptor
 {
+    protected function describeDefaults(array $options = array())
+    {
+        $data['builtin_form_types'] = $this->getCoreTypes();
+        $data['services_form_types'] = array_values(array_diff($options['types'], $data['builtin_form_types']));
```

Comment:
builtin types will be all from `$options['types']` within `Symfony\Component\Form\Extension\Core\Type\` namespace no? Isnt that more easy to do instead of `getCoreTypes` hack.

Reply:
Also i prefer services types to be shown above built in ones. Personal pref though.

Reply:
> builtin types will be all from `$options['types']` within `Symfony\Component\Form\Extension\Core\Type\ namespace` no?

Nope, only those registered, the most of them aren't services.

Reply:
Got it. Perhaps add static getter for that in FormRegistry, as little new feat. But reflection is fine as well i guess xD

Reply:
> Perhaps add static getter for that in FormRegistry, as little new feat.

Not sure how to do that, because types is concern of `FormExtensionInterface` instances and `loadTypes` method or `types` property (both inaccessible directly) so would need hack it anyway.

Reply:
My idea was `FormRegistry::getCoreTypes` with the implem. of `CoreExtension::loadTypes` which in turn does simply `return FormRegistry::getCoreTypes();`.

For this PR you'd use `FormRegistry::getCoreTypes()` as well. `CoreExtension::getTypes` might do as well :)

Reply:
Yep, still not sure if worth it just for debug purpose.

Reply:
True :+1: